test/OpenTelemetry.Contrib.Instrumentation.Quartz.Tests/QuartzDiagnosticListenerTests.cs

```diff
@@ -0,0 +1,397 @@
+
+            barrier.SignalAndWait(TestTimeout);
+
+            await scheduler.Shutdown(true);
+
+            // Assert
+            Assert.Equal(3, activityProcessor.Invocations.Count);
+            var activity = (Activity)activityProcessor.Invocations[1].Arguments[0];
+
+            Assert.Equal("Quartz.Job.Execute", activity.OperationName);
+            Assert.Equal(ActivityKind.Internal, activity.Kind);
+            Assert.Equal("Scheduler", activity.Tags.SingleOrDefault(t => t.Key.Equals("scheduler.name")).Value);
+            Assert.Equal(SchedulerConstants.DefaultGroup, activity.Tags.SingleOrDefault(t => t.Key.Equals("job.group")).Value);
+            Assert.Equal(SchedulerConstants.DefaultGroup, activity.Tags.SingleOrDefault(t => t.Key.Equals("trigger.group")).Value);
+            Assert.Equal(testId, activity.Tags.SingleOrDefault(t => t.Key.Equals("test.id")).Value);
+        }
+
+        [Fact]
+        public async Task Should_Record_Exception_When_Record_Exception_Enabled()
+        {
+            // Arrange
+            Barrier barrier = new Barrier(2);
+            List<DateTime> jobExecTimestamps = new List<DateTime>();
+
+            var activityProcessor = new Mock<BaseProcessor<Activity>>();
+
+            using var tel = Sdk.CreateTracerProviderBuilder()
+                .SetSampler(new AlwaysOnSampler())
+                .AddQuartzInstrumentation(q =>
+                    q.RecordException = true)
+                .AddProcessor(activityProcessor.Object)
+                .Build();
+
+            var schedulerConfig = SchedulerBuilder.Create("AUTO", "Scheduler");
+            schedulerConfig.UseDefaultThreadPool(x => x.MaxConcurrency = 10);
+            var scheduler = await schedulerConfig.BuildScheduler();
+
+            scheduler.Context.Put("BARRIER", barrier);
+            scheduler.Context.Put("DATESTAMPS", jobExecTimestamps);
+            await scheduler.Start();
+
+            var testId = Guid.NewGuid().ToString();
+            JobDataMap jobDataMap = new JobDataMap { { "TestId", testId } };
+
+            var name = Guid.NewGuid().ToString();
+            var job = JobBuilder.Create<TestJobExecutionExceptionJob>()
+                .WithIdentity(name, SchedulerConstants.DefaultGroup)
+                .UsingJobData(jobDataMap)
+                .Build();
+
+            var trigger = TriggerBuilder.Create()
+                .WithIdentity(name, SchedulerConstants.DefaultGroup)
+                .StartNow()
+                .Build();
+
+            // Act
+            await scheduler.ScheduleJob(job, trigger);
+
+            barrier.SignalAndWait(TimeSpan.FromSeconds(1));
+
+            await scheduler.Shutdown(true);
+
+            // Assert
+            Assert.Equal(3, activityProcessor.Invocations.Count);
+            var activity = (Activity)activityProcessor.Invocations[1].Arguments[0];
+
+            Assert.Equal("ERROR", activity.Tags.SingleOrDefault(t => t.Key.Equals(SpanAttributeConstants.StatusCodeKey)).Value);
```

Comment:
when RecordException is true, please validate that the exception are stored as activityevent. (the activity.RecordException does just that, so would be good to validate that in unit tests)

Reply:
added assertion for `ActivityEvent.Name`  name